### ISSUE TITLE
New MIDI file for midi opcode examples (and replace FLTK examples)

### ISSUE THREAD

**Issue by AsterixMusic** [csound/manual#560]
most of the examples that use FLTK are midi opcodes.
I suggest to have 1 standard midi-file with "all" kind of message, with related list.
Ex: 
channel  1: 4 notes with different velocity
channel 2: 1 note with control change
channel 3: 2 control change
channel 4: 2 notes - 1 program change - 2 notes
channel 5: .....
........
........

This file could be useful to test also all the MIDI opcodes.
[midi_Csound.mid.zip](https://github.com/csound/manual/files/6796566/midi_Csound.mid.zip)

**Comment by csounder** [comment]
excellent idea.
- can you share a .csd file that works currently with this proposed
standard MIDI file.

*- dB*


*Dr. Richard Boulanger*

Professor

Electronic Production and Design

*Berklee College of Music*

Professional Writing & Technology Division

**Comment by AsterixMusic** [comment]
Not yet, me and Menno are still looking for all the data we need to put on this MIDI file:

Polyphonic aftertouch
sustain
pitch bend
…
...
I’m thinking about 1 big example with many instruments, one instruments every message.

Ex:

Instr 1: fey notes
Instr 2: control change
Instr 3: polyphonic aftertouch
Instr 4: Program change
Instr 5: …
Instr 6: …

Do you think it’s a good idea, or it’s better to think about many examples


-------------------------------------------------
Stefano Cucchi
I.T.B. Project Studio
Viale Europa 15, 20088 Rosate (Mi)
Tel: 3402601621
www.itbprojectstudio.com
Mail: ***@***.***

**Comment by csounder** [comment]
Yes, this is an excellent plan, with one larger .csd file that shows how
(in individual instruments) each of these MIDI messages might be
interpreted, mapped, used.  This would make a great "supermodel" for
students especially.

*- dB*


*Dr. Richard Boulanger*

Professor

Electronic Production and Design

*Berklee College of Music*

Professional Writing & Technology Division

**Comment by tjingboem** [comment]
the idea of having 1 'big' Midi file that should cover the replacement of all FLTK Midi instruments has yet another advantage; if we find, along the way, that a new CC or sysex should be added because an example requires this due to the particular Midi opcode, we can simply add this addition to an existing Midi channel or create a new Midi channel for it. And then save it under the same name.
If we re-use one and the same Midi file name for such changes, we do not need to get back to already existing Midi FLTK replaced examples.

Each new Midi example should be able to extract its useful information from that channel. I believe Midi Out can have 64 channels, Midi In has 16? Not sure...
